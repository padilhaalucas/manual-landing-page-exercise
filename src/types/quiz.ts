export interface QuizOption {
  display: string;
  value: string | boolean;
  isRejection: boolean;
}

export interface QuizQuestion {
  question: string;
  type: string;
  options: QuizOption[];
}

export interface QuizData {
  questions: QuizQuestion[];
}