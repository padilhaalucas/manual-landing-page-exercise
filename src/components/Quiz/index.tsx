import Image from 'next/image'
import Link from 'next/link'
import React, { useCallback, useEffect, useRef, useState } from 'react';

import { useQuiz } from '@/contexts/quiz.context';
import quizData from '@/mocked-data-source/quiz.json';

const Quiz: React.FC = () => {
  const {
    currentQuestion,
    answers,
    showQuiz,
    quizCompleted,
    setCurrentQuestion,
    setAnswers,
    setShowQuiz,
    setQuizCompleted,
    isRejected
  } = useQuiz();

  const modalRef = useRef<HTMLDivElement>(null);
  const [fade, setFade] = useState(false);

  const handleAnswer = (answer: string) => {
    setFade(true);
    setTimeout(() => {
      const newAnswers = [...answers, answer];
      setAnswers(newAnswers);

      if (currentQuestion < quizData.questions.length - 1) {
        setCurrentQuestion(currentQuestion + 1);
      } else {
        setQuizCompleted(true);
      }
      setFade(false);
    }, 300);
  };

  const handleBack = () => {
    if (currentQuestion > 0) {
      setCurrentQuestion(currentQuestion - 1);
      setAnswers(answers.slice(0, -1));
    }
  };

  const closeQuiz = useCallback(() => {
    setShowQuiz(false);
    setTimeout(() => {
      setQuizCompleted(false);
      setCurrentQuestion(0);
      setAnswers([]);
    }, 300);
  }, [setShowQuiz, setQuizCompleted, setCurrentQuestion, setAnswers]);

  const question = quizData.questions[currentQuestion];

  useEffect(() => {
    const handleClickOutside = (event: MouseEvent) => {
      if (modalRef.current && !modalRef.current.contains(event.target as Node)) {
        closeQuiz();
      }
    };

    if (showQuiz) {
      document.addEventListener('mousedown', handleClickOutside);
      document.body.style.overflow = 'hidden';
    }

    return () => {
      document.removeEventListener('mousedown', handleClickOutside);
      document.body.style.overflow = '';
    };
  }, [showQuiz, closeQuiz]);

  return (
    <div
      data-testid="quiz-container"
      className={`
        fixed inset-0 z-50 flex items-center justify-center p-4 transition-all duration-300 ease-in-out
        ${showQuiz ? 'visible bg-black bg-opacity-50' : 'invisible bg-black bg-opacity-0'}
      `}
    >
      <div
        data-testid="quiz-modal"
        ref={modalRef}
        className={`
          bg-white rounded-lg shadow-xl w-full max-w-md overflow-x-hidden max-h-[40rem] transition-all duration-300 ease-in-out
          ${showQuiz ? 'opacity-100 scale-100 translate-y-0' : 'opacity-0 scale-95 translate-y-8'}
        `}
      >
        {/* Header */}
        <div className="bg-sage-green p-6 border-b border-b-gray-200 flex justify-between items-center">
          <h2 className="text-2xl font-bold text-dark-green-manual">Hair Loss Quiz</h2>
          <button onClick={closeQuiz} className="text-dark-green-manual">
            <Image src={"/assets/close-icon.svg"} alt={"Close Icon"} width={24} height={24} />
          </button>
        </div>

        {/* Content */}
        <div className="p-6">
          {quizCompleted ? (
            <div
              data-testid="quiz-transition-element"
              className="text-center transition-all duration-300 ease-in-out opacity-100 translate-y-0"
            >
              {isRejected && (
                <>
                  <h3 className="text-2xl font-bold mb-4 text-dark-green-manual">
                    Unfortunately, we are unable to prescribe this medication for you.
                  </h3>
                  <p className="mb-6 text-dark-green-manual/80">
                    This is because finasteride can alter the PSA levels, which may be used to monitor for cancer.
                    You should discuss this further with your GP or specialist if you would still like this medication.
                  </p>
                </>
              )}
              {!isRejected && (
                <>
                  <h3 className="text-4xl font-bold mb-4 text-dark-green-manual">
                    Great news!
                  </h3>
                  <h4 className="text-2xl font-bold mb-4 text-dark-green-manual">We have the perfect treatment for your hair loss.</h4>
                  <p className="mb-6 text-dark-green-manual/80">
                    Proceed to <Link className="text-blue-500 font-bold" href="https://manual.co" target="_blank">www.manual.co</Link>, and prepare to say hello to your new hair!
                  </p>
                </>
              )}

              <button
                className="bg-dark-green-manual text-white px-6 py-2 rounded hover:bg-opacity-90 active:bg-opacity-100 transition-all duration-300"
                onClick={closeQuiz}
              >
                Close Quiz
              </button>
            </div>
          ) : (
            <div className={`transition-opacity duration-300 ease-in-out ${fade ? 'opacity-0' : 'opacity-100'}`}>
              <h3 className="text-xl font-semibold mb-4 text-dark-green-manual">{question.question}</h3>
              <div className="space-y-4">
                {question.options.map((option, index) => (
                  <button
                    key={index}
                    className={`
                      flex justify-center items-center w-full p-4 border border-gray-200 rounded-lg
                      text-dark-green-manual hover:bg-sage-green-manual hover:border-dark-green-manual transition-all duration-300
                    `}
                    onClick={() => handleAnswer(option.value as string)}
                  >
                    {option.display.includes('<img') ? (
                      <div className="flex flex-row">
                        <span className="absolute left-10 font-black text-3xl text-light-blue-manual tracking-tight">
                          {index + 1}
                        </span>
                        <div dangerouslySetInnerHTML={{ __html: option.display }} />
                      </div>
                    ) : (
                      option.display
                    )}
                  </button>
                ))}
              </div>
            </div>
          )}
        </div>

        {/* Footer */}
        {!quizCompleted && (
          <div className="bg-gray-100 px-6 py-4 flex flex-col justify-between items-start">
            <div className="text-sm text-dark-green-manual/80">
              Question <strong>{currentQuestion + 1}</strong> of {quizData.questions.length}
            </div>
            {currentQuestion > 0 && (
              <button
                className="text-dark-green-manual hover:text-opacity-75 font-semibold"
                onClick={handleBack}
              >
                Back
              </button>
            )}
          </div>
        )}
      </div>
    </div>
  );
};

export default Quiz;