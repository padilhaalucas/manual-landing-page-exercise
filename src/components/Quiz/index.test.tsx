import Image from 'next/image'
import React from 'react';
import { render, screen, fireEvent, act } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Quiz from './index';
import { QuizProvider, QuizContext, QuizContextType } from '@/contexts/quiz.context';
import quizData from '@/mocked-data-source/quiz.json';
import { QuizData } from '@/types/quiz';

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    return <Image {...props} alt="test-img-quiz" />
  },
}));

const mockQuizData: QuizData = {
  questions: quizData.questions
};

const mockQuizContextValue: QuizContextType = {
  quizData: mockQuizData,
  currentQuestion: 0,
  answers: [],
  showQuiz: true,
  quizCompleted: false,
  setCurrentQuestion: jest.fn(),
  setAnswers: jest.fn(),
  setShowQuiz: jest.fn(),
  setQuizCompleted: jest.fn(),
  isRejected: false,
};

const renderWithQuizProvider = (ui: React.ReactElement, contextValue = mockQuizContextValue) => {
  return render(
    <QuizProvider initialData={mockQuizData}>
      <QuizContext.Provider value={contextValue}>
        {ui}
      </QuizContext.Provider>
    </QuizProvider>
  );
};

describe('Components/Quiz', () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should render the quiz header', () => {
    renderWithQuizProvider(<Quiz />);
    expect(screen.getByText('Hair Loss Quiz')).toBeInTheDocument();
  });

  it('should set body overflow to hidden when quiz is shown', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      showQuiz: true,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);

    expect(document.body.style.overflow).toBe('hidden');
  });

  it('should render the first question', () => {
    renderWithQuizProvider(<Quiz />);
    expect(screen.getByText(mockQuizData.questions[0].question)).toBeInTheDocument();
  });

  it('should render all options for the first question', () => {
    renderWithQuizProvider(<Quiz />);
    mockQuizData.questions[0].options.forEach((option) => {
      expect(screen.getByAltText(option.value as string)).toBeInTheDocument();
    });
  });

  it('should handle answer selection', async () => {
    const mockSetAnswers = jest.fn();
    const mockSetCurrentQuestion = jest.fn();
    const customContextValue = {
      ...mockQuizContextValue,
      setAnswers: mockSetAnswers,
      setCurrentQuestion: mockSetCurrentQuestion,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    const firstOption = screen.getByAltText(mockQuizData.questions[0].options[0].value as string);
    fireEvent.click(firstOption);

    act(() => {
      jest.advanceTimersByTime(300);
    });

    expect(mockSetAnswers).toHaveBeenCalled();
    expect(mockSetCurrentQuestion).toHaveBeenCalled();
  });

  it('should render the "Back" button after the first question', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      currentQuestion: 1,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    expect(screen.getByText('Back')).toBeInTheDocument();
  });

  it('should handle quiz completion', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      quizCompleted: true,
      isRejected: false,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    expect(screen.getByText('Great news!')).toBeInTheDocument();
    expect(screen.getByText('We have the perfect treatment for your hair loss.')).toBeInTheDocument();
  });

  it('should handle rejection', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      quizCompleted: true,
      isRejected: true,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    expect(screen.getByText('Unfortunately, we are unable to prescribe this medication for you.')).toBeInTheDocument();
  });

  it('should handle going back to the previous question', () => {
    const mockSetCurrentQuestion = jest.fn();
    const mockSetAnswers = jest.fn();
    const customContextValue = {
      ...mockQuizContextValue,
      currentQuestion: 1,
      answers: ['answer1'],
      setCurrentQuestion: mockSetCurrentQuestion,
      setAnswers: mockSetAnswers,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    const backButton = screen.getByText('Back');
    fireEvent.click(backButton);

    expect(mockSetCurrentQuestion).toHaveBeenCalledWith(0);
    expect(mockSetAnswers).toHaveBeenCalledWith([]);
  });

  it('should handle completing the quiz', () => {
    const mockSetQuizCompleted = jest.fn();
    const mockSetCurrentQuestion = jest.fn();
    const customContextValue = {
      ...mockQuizContextValue,
      currentQuestion: quizData.questions.length - 1,
      setQuizCompleted: mockSetQuizCompleted,
      setCurrentQuestion: mockSetCurrentQuestion,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    const lastOption = screen.getByText(mockQuizData.questions[quizData.questions.length - 1].options[0].display as string);
    fireEvent.click(lastOption);

    act(() => {
      jest.advanceTimersByTime(300);
    });

    expect(mockSetQuizCompleted).toHaveBeenCalledWith(true);
    expect(mockSetCurrentQuestion).not.toHaveBeenCalled();
  });

  it('should handle closing the quiz', () => {
    const mockSetShowQuiz = jest.fn().mockImplementation(() => {
      document.body.style.overflow = '';
    });
    const mockSetQuizCompleted = jest.fn();
    const mockSetCurrentQuestion = jest.fn();
    const mockSetAnswers = jest.fn();
    const customContextValue = {
      ...mockQuizContextValue,
      showQuiz: true,
      setShowQuiz: mockSetShowQuiz,
      setQuizCompleted: mockSetQuizCompleted,
      setCurrentQuestion: mockSetCurrentQuestion,
      setAnswers: mockSetAnswers,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    expect(document.body.style.overflow).toBe('hidden');

    const closeButton = screen.getByAltText('Close Icon');
    fireEvent.click(closeButton);

    expect(mockSetShowQuiz).toHaveBeenCalledWith(false);
    expect(document.body.style.overflow).toBe('');

    act(() => {
      jest.advanceTimersByTime(300);
    });

    expect(mockSetQuizCompleted).toHaveBeenCalledWith(false);
    expect(mockSetCurrentQuestion).toHaveBeenCalledWith(0);
    expect(mockSetAnswers).toHaveBeenCalledWith([]);
  });

  it('should handle click outside to close the quiz', () => {
    const mockSetShowQuiz = jest.fn();
    const customContextValue = {
      ...mockQuizContextValue,
      setShowQuiz: mockSetShowQuiz,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    fireEvent.mouseDown(document.body);
    expect(document.body.style.overflow).toBe('hidden');

    expect(mockSetShowQuiz).toHaveBeenCalledWith(false);
  });

  it('should not close the quiz when clicking inside', () => {
    const mockSetShowQuiz = jest.fn();
    const customContextValue = {
      ...mockQuizContextValue,
      setShowQuiz: mockSetShowQuiz,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    const quizContent = screen.getByText('Hair Loss Quiz').parentElement;
    fireEvent.mouseDown(quizContent!);

    expect(mockSetShowQuiz).not.toHaveBeenCalled();
  });

  it('should handle non-image options', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      currentQuestion: 1, // Assuming the second question has non-image options
    };

    renderWithQuizProvider(<Quiz />, customContextValue);
    mockQuizData.questions[1].options.forEach((option) => {
      expect(screen.getByText(option.display)).toBeInTheDocument();
    });
  });

  it('should apply correct classes when showQuiz is true', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      showQuiz: true,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);

    const quizContainer = screen.getByTestId('quiz-container'); // Using data-testid
    expect(quizContainer).toHaveClass('visible', 'bg-black', 'bg-opacity-50');

    const modal = screen.getByTestId('quiz-modal'); // Using data-testid
    expect(modal).toHaveClass('opacity-100', 'scale-100', 'translate-y-0');
  });

  it('should apply correct classes when showQuiz is false', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      showQuiz: false,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);

    const quizContainer = screen.getByTestId('quiz-container'); // Using data-testid
    expect(quizContainer).toHaveClass('invisible', 'bg-black', 'bg-opacity-0');

    const modal = screen.getByTestId('quiz-modal'); // Using data-testid
    expect(modal).toHaveClass('opacity-0', 'scale-95', 'translate-y-8');
  });

  it('should apply correct classes when quizCompleted is true', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      quizCompleted: true,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);

    const quizContent = screen.getByText('Great news!').parentElement;
    expect(quizContent).toHaveClass('opacity-100', 'translate-y-0');
  });

  it('should apply correct classes when quizCompleted is false', () => {
    const customContextValue = {
      ...mockQuizContextValue,
      quizCompleted: false,
    };

    renderWithQuizProvider(<Quiz />, customContextValue);

    const questionElement = screen.getByText(mockQuizData.questions[0].question).parentElement;
    expect(questionElement).toHaveClass('transition-opacity', 'duration-300', 'ease-in-out', 'opacity-100');
  });

  it('should apply correct classes for quiz completion transition', () => {
    // Setup initial context with quizCompleted true
    const customContextValue = {
      ...mockQuizContextValue,
      quizCompleted: true,
    };

    const { rerender } = renderWithQuizProvider(<Quiz />, customContextValue);

    // Check the element classes when quizCompleted is true
    let completionElement = screen.getByText('Great news!').parentElement;
    expect(completionElement).toHaveClass('opacity-100', 'translate-y-0');

    // Update the context to quizCompleted false
    customContextValue.quizCompleted = true;
    rerender(
      <QuizProvider initialData={mockQuizData}>
        <QuizContext.Provider value={customContextValue}>
          <Quiz />
        </QuizContext.Provider>
      </QuizProvider>
    );

    // Ensure we target the right parent element for transition check
    completionElement = screen.getByTestId('quiz-transition-element');
    expect(completionElement).toHaveClass('opacity-100', 'translate-y-0');
  });

  it('should match snapshot', () => {
    const tree = renderer.create(
      <QuizProvider initialData={mockQuizData}>
        <QuizContext.Provider value={mockQuizContextValue}>
          <Quiz />
        </QuizContext.Provider>
      </QuizProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

