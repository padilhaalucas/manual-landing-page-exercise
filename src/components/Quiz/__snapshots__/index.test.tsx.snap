// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Quiz should match snapshot 1`] = `
<div
  className="
        fixed inset-0 z-50 flex items-center justify-center p-4 transition-all duration-300 ease-in-out
        visible bg-black bg-opacity-50
      "
  data-testid="quiz-container"
>
  <dialog
    className="
          bg-white rounded-lg shadow-xl w-full max-w-md overflow-x-hidden max-h-[40rem] transition-all duration-300 ease-in-out
          opacity-100 scale-100 translate-y-0
        "
    data-testid="quiz-modal"
  >
    <div
      className="bg-sage-green p-6 border-b border-b-gray-200 flex justify-between items-center"
    >
      <h2
        className="text-2xl font-bold text-dark-green-manual"
      >
        Hair Loss Quiz
      </h2>
      <button
        className="text-dark-green-manual"
        onClick={[Function]}
      >
        <img
          alt="Close Icon"
          height={24}
          src="/assets/close-icon.svg"
          width={24}
        />
      </button>
    </div>
    <div
      className="p-6"
    >
      <div
        className="transition-opacity duration-300 ease-in-out opacity-100"
      >
        <h3
          className="text-xl font-semibold mb-4 text-dark-green-manual"
        >
          Which image best matches your hair loss?
        </h3>
        <div
          className="space-y-4"
        >
          <button
            className="
                      flex justify-center items-center w-full p-4 border border-gray-200 rounded-lg
                      text-dark-green-manual hover:bg-sage-green-manual hover:border-dark-green-manual transition-all duration-300
                    "
            onClick={[Function]}
          >
            <div
              className="flex flex-row"
            >
              <span
                className="absolute left-10 font-black text-3xl text-light-blue-manual tracking-tight"
              >
                1
              </span>
              <div
                dangerouslySetInnerHTML={
                  {
                    "__html": "<img alt="Temples" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/temples-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/temples-hairloss%402x.png 2x" />",
                  }
                }
              />
            </div>
          </button>
          <button
            className="
                      flex justify-center items-center w-full p-4 border border-gray-200 rounded-lg
                      text-dark-green-manual hover:bg-sage-green-manual hover:border-dark-green-manual transition-all duration-300
                    "
            onClick={[Function]}
          >
            <div
              className="flex flex-row"
            >
              <span
                className="absolute left-10 font-black text-3xl text-light-blue-manual tracking-tight"
              >
                2
              </span>
              <div
                dangerouslySetInnerHTML={
                  {
                    "__html": "<img alt="Temples & Crown" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/templescrown-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/templescrown-hairloss%402x.png 2x"/>",
                  }
                }
              />
            </div>
          </button>
          <button
            className="
                      flex justify-center items-center w-full p-4 border border-gray-200 rounded-lg
                      text-dark-green-manual hover:bg-sage-green-manual hover:border-dark-green-manual transition-all duration-300
                    "
            onClick={[Function]}
          >
            <div
              className="flex flex-row"
            >
              <span
                className="absolute left-10 font-black text-3xl text-light-blue-manual tracking-tight"
              >
                3
              </span>
              <div
                dangerouslySetInnerHTML={
                  {
                    "__html": "<img alt="Patchy" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/patchy-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/patchy-hairloss%402x.png 2x"/>",
                  }
                }
              />
            </div>
          </button>
          <button
            className="
                      flex justify-center items-center w-full p-4 border border-gray-200 rounded-lg
                      text-dark-green-manual hover:bg-sage-green-manual hover:border-dark-green-manual transition-all duration-300
                    "
            onClick={[Function]}
          >
            <div
              className="flex flex-row"
            >
              <span
                className="absolute left-10 font-black text-3xl text-light-blue-manual tracking-tight"
              >
                4
              </span>
              <div
                dangerouslySetInnerHTML={
                  {
                    "__html": "<img alt="Moderate" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/moderate-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/moderate-hairloss%402x.png 2x" />",
                  }
                }
              />
            </div>
          </button>
          <button
            className="
                      flex justify-center items-center w-full p-4 border border-gray-200 rounded-lg
                      text-dark-green-manual hover:bg-sage-green-manual hover:border-dark-green-manual transition-all duration-300
                    "
            onClick={[Function]}
          >
            <div
              className="flex flex-row"
            >
              <span
                className="absolute left-10 font-black text-3xl text-light-blue-manual tracking-tight"
              >
                5
              </span>
              <div
                dangerouslySetInnerHTML={
                  {
                    "__html": "<img alt="Extensive" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/extensive-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/extensive-hairloss%402x.png 2x"/>",
                  }
                }
              />
            </div>
          </button>
          <button
            className="
                      flex justify-center items-center w-full p-4 border border-gray-200 rounded-lg
                      text-dark-green-manual hover:bg-sage-green-manual hover:border-dark-green-manual transition-all duration-300
                    "
            onClick={[Function]}
          >
            <div
              className="flex flex-row"
            >
              <span
                className="absolute left-10 font-black text-3xl text-light-blue-manual tracking-tight"
              >
                6
              </span>
              <div
                dangerouslySetInnerHTML={
                  {
                    "__html": "<img alt="Complete" src="https://s3-eu-west-1.amazonaws.com/manualco/questions/complete-hairloss.png" srcset="https://s3-eu-west-1.amazonaws.com/manualco/questions/complete-hairloss%402x.png 2x" />",
                  }
                }
              />
            </div>
          </button>
        </div>
      </div>
    </div>
    <div
      className="bg-gray-100 px-6 py-4 flex flex-col justify-between items-start"
    >
      <div
        className="text-sm text-dark-green-manual/80"
      >
        Question 
        <strong>
          1
        </strong>
         of 
        3
      </div>
    </div>
  </dialog>
</div>
`;
